islandora_solr_config/includes/rss_results.inc

```diff
@@ -191,7 +190,85 @@ class IslandoraSolrResultsRSS extends IslandoraSolrResults {
     $result['url'] = $base_url;
     $result['description'] = t('Aggregated search results of: @query', array('@query' => $query));
     $result['langcode'] = NULL;
+    $feed_url = request_uri();
+    $page_params = drupal_get_query_parameters();
+
+    if (!isset($page_params['page'])) {
+      $page_number = '0';
+    }
+    else {
+      $page_number = $page_params['page'];
+    }
+    if (isset($page_params['page'])) {
+      $first_params = $page_params;
+      unset($first_params['page']);
+      $link_first = url(
+        request_path(),
+        array(
+          'absolute' => TRUE,
+          'query' => $first_params,
+        )
+      );
+    }
+    else {
+      $link_first = $base_url . $feed_url;
+    }
+
+    $page_next = $page_number + 1;
+    if ($page_number > 0) {
+      $page_prev = $page_number - 1;
+    }
+
+    $next_page = array(
+      'query' => array(
+        'page' => $page_next,
+      )+$page_params,
+    );
+
+    // Only set Next link if there is a next page.
+    $rss_limit = variable_get('islandora_solr_config_rss_limit', 50);
+    if (count($docs) == $rss_limit) {
+      $link_next = url($base_url . "/" . request_path(), $next_page);
```

Comment:
Like to see this use the absolute argument instead of $base_url

Reply:
I've tried with the absolute argument; I still get a relative path.

Reply:
Thats strange, when I use absolute I get a full path, and the d.o page says it gives the full path. 

```php
url(request_path(), array('absolute' => TRUE))
```

is always returning a full URI to me.

Reply:
`$link_next = url(request_path(), $next_page, array('absolute' => TRUE));`
produces
`<link rel="next">/islandora/search/test?page=1&type=dismax&islandora_solr_search_navigation=0&solr_profile=rss</link>`

`$link_next = url(request_path(), array('absolute' => TRUE), $next_page);` 
returns 
` <link rel="next">http://localhost:8000/islandora/search/test</link>`

Hmm...

`$link_next = url(url(request_path(), array('absolute' => TRUE)), $next_page);`
returns
`<link rel="next">http://localhost:8000/islandora/search/test?page=1&type=dismax&islandora_solr_search_navigation=0&solr_profile=rss</link>`

So what I need is nested `url()` functions, I guess?

Reply:
No you just need to pass all the parameters in a single array. 

This is one way to do it:
```php
url(request_path(), array('absolute' => TRUE) + $next_page);
```

Or you could directly put it in the definition of `$next_page`, which really should be named `$args` or something like that.

```php
 $args = array(
  'absolute' => TRUE,
  'query' => array(
    'page' => $page_next,
  )+$page_params,
);
url(request_path(), $args);
```

or you could even collapse the arg definition into the function call: 

```php
url(request_path(), array(
  'absolute' => TRUE,
  'query' => array(
    'page' => $page_next,
  )+$page_params,
));
```

Reply:
Ha, OK, that worked, thanks